### PR TITLE
`SpotifyBlockComponent` as an island

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
Moves `SpotifyBlockComponent` over to the island pattern

## Why?
This work forms part of the larger move away from `App.tsx`

### CODE REVIEW

File: dotcom-rendering/src/web/lib/renderElement.tsx [hunk 1]
```diff
@@ -514,13 +514,7 @@ export const renderElement = ({
...
 			return [
 				true,
-				<ClickToView
-					role={element.role}
-					isTracking={element.isThirdPartyTracking}
-					isMainMedia={isMainMedia}
-					source={element.source}
-					sourceDomain={element.sourceDomain}
-				>
+				<Island deferUntil="visible">
```
Comment (jamesgorrie) [inline dotcom-rendering/src/web/lib/renderElement.tsx:517]: Whats the reason for moving ClickToView into `SpotifyBlockContent`? Is it suggesting that we always click to view the `SpotifyBlockContent` and blocks like it?

**Review by jamesgorrie — COMMENTED** [review]
A few questions mainly for my learning.

File: dotcom-rendering/src/web/components/SpotifyBlockComponent.importable.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	title?: string;
+	format: ArticleFormat;
+	caption?: string;
+	credit?: string;
+	role?: RoleType;
+	isTracking: boolean;
+	isMainMedia: boolean;
+	source?: string;
+	sourceDomain?: string;
+}> = ({
```
[the remark this answers is not in the document]
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/SpotifyBlockComponent.importable.tsx:23]: I am fully on board with not using FC. Here's [another good write up](https://kentcdodds.com/blog/how-to-write-a-react-component-in-typescript) that talks about this. Happy to migrate this older code to the newer conventions.

File: dotcom-rendering/src/web/lib/renderElement.tsx (hunk 1 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/web/lib/renderElement.tsx:517]: Yes, basically it is not only technically required now but, as [Olly pointed out on another PR](https://github.com/guardian/dotcom-rendering/pull/3824#issuecomment-1011022614) the other day, it also helps firm up the legal requirement that it must always be used. With it integral to the component, you can't accidentally forget it.

Comment (jamesgorrie) [reply dotcom-rendering/src/web/lib/renderElement.tsx:517]: > the legal requirement that it must always be used

Great reason.

**Review by jamesgorrie — APPROVED** [review]

**Comment by jamesgorrie** [comment]
part of https://github.com/guardian/dotcom-rendering/pull/3629

**Pull request merged**